### PR TITLE
terraform-providers.checkly: init at 0.7.1

### PR DESCRIPTION
**Pull request by mmahut**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mmahut** [comment]
@GrahamcOfBorg build terraform-providers.checkly

**Comment by mmahut** [comment]
@GrahamcOfBorg build terraform-providers.checkly

File: pkgs/applications/networking/cluster/terraform-providers/default.nix [hunk 1]
```diff
@@ -55,6 +55,7 @@ let
 
     # Packages that don't fit the default model
     ansible = callPackage ./ansible {};
+    checkly = callPackage ./checkly {};
```
Comment (kalbasit) [inline pkgs/applications/networking/cluster/terraform-providers/default.nix:58]: Why doesn't this fit the default model? I don't see anything specific.

**Review by kalbasit — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/cluster/terraform-providers/default.nix (hunk 1 above)
Comment (mmahut) [reply pkgs/applications/networking/cluster/terraform-providers/default.nix:58]: Not sure, when I tried it did not build.

Comment (mmahut) [reply pkgs/applications/networking/cluster/terraform-providers/default.nix:58]:
```building
go/src/github.com/checkly/terraform-provider-checkly/checkly/provider.go:7:2: cannot find package "github.com/checkly/checkly-go-sdk" in any of:
        /nix/store/qvxsh2nxjf598mzfi6hl9dy4qv3698x4-go-1.15.6/share/go/src/github.com/checkly/checkly-go-sdk (from $GOROOT)
        /build/go/src/github.com/checkly/checkly-go-sdk (from $GOPATH)
```

I'm not sure how to mitigate that.

File: pkgs/applications/networking/cluster/terraform-providers/checkly/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1wv7zdzzg8cjcz5h8vvnaznnbp8r7kbdc9hj10s8dmnmqkf9rf83";
+  };
+
+  vendorSha256 = "1k4rpin0ijs31hlfcrgyz97yll89ff6lbnkkscyfiw61xcsz6mhm";
+
+  postInstall = "mv $out/bin/terraform-provider-checkly{,_v${version}}";
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/terraform-providers/checkly/default.nix:18]:
```suggestion
  meta = with lib; {
```

File: pkgs/applications/networking/cluster/terraform-providers/checkly/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromGitHub {
+    owner = "checkly";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1wv7zdzzg8cjcz5h8vvnaznnbp8r7kbdc9hj10s8dmnmqkf9rf83";
+  };
+
+  vendorSha256 = "1k4rpin0ijs31hlfcrgyz97yll89ff6lbnkkscyfiw61xcsz6mhm";
+
+  postInstall = "mv $out/bin/terraform-provider-checkly{,_v${version}}";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/terraform-providers/checkly/default.nix:16]:
```suggestion
  postInstall = ''
    mv $out/bin/terraform-provider-checkly{,_v${version}}
  '';
```

File: pkgs/applications/networking/cluster/terraform-providers/default.nix (hunk 1 above)
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/terraform-providers/default.nix:58]:
```suggestion
    checkly = callPackage ./checkly { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Can this use the generic terraform template?

**Comment by roberth** [comment]
> Can this use the generic terraform template?

Apparently not, considering previous comments.

Lgtm, but ofborg failed. Need to bring `lib` into scope.

**Comment by zowoq** [comment]
The current version does build with the generic terraform style.

```json
  "checkly": {
    "owner": "checkly",
    "provider-source-address": "registry.terraform.io/checkly/checkly",
    "repo": "terraform-provider-checkly",
    "rev": "v1.3.0",
    "sha256": "11vwl983lh983c1x3f7zc8b7i9f4pymk4j1ikf04vz2kgxry49yr",
    "vendorSha256": "14wp5sa2fm3hlmyacfy4aacx2cz5bxf24r4fjwd6f2hskzjz6825",
    "version": "1.3.0"
  },
```

**Comment by 1000101** [comment]
> The current version does build with the generic terraform style.
> 
> ```json
>   "checkly": {
>     "owner": "checkly",
>     "provider-source-address": "registry.terraform.io/checkly/checkly",
>     "repo": "terraform-provider-checkly",
>     "rev": "v1.3.0",
>     "sha256": "11vwl983lh983c1x3f7zc8b7i9f4pymk4j1ikf04vz2kgxry49yr",
>     "vendorSha256": "14wp5sa2fm3hlmyacfy4aacx2cz5bxf24r4fjwd6f2hskzjz6825",
>     "version": "1.3.0"
>   },
> ```



> The current version does build with the generic terraform style.
> 
> ```json
>   "checkly": {
>     "owner": "checkly",
>     "provider-source-address": "registry.terraform.io/checkly/checkly",
>     "repo": "terraform-provider-checkly",
>     "rev": "v1.3.0",
>     "sha256": "11vwl983lh983c1x3f7zc8b7i9f4pymk4j1ikf04vz2kgxry49yr",
>     "vendorSha256": "14wp5sa2fm3hlmyacfy4aacx2cz5bxf24r4fjwd6f2hskzjz6825",
>     "version": "1.3.0"
>   },
> ```

With(out) your permission, hope you don't mind :) https://github.com/NixOS/nixpkgs/pull/153336

**Pull request closed without merging**